src/link_time_code_gen.mli

```diff
@@ -0,0 +1,23 @@
+(** {1 Handle link time code generation} *)
+
+type libs_or_module =
+  | Libs of Lib.L.t
+  | Module of Module.t
+(** List of library and modules in link order *)
+
+type t = {
+  libs_or_module: libs_or_module list;
+  flags: Ocaml_flags.t;
+}
+
+
+val create: ?flags:Ocaml_flags.t -> Lib.L.t -> t
```

Comment:
I think we can do everything in one step. From the outside of this module, `findlib.dynload` is basically just an implementation detail.

BTW, for multi-line function signatures, we use the following style in the code base:

```ocaml
val foo
  :  x
  -> y
  -> z
```